### PR TITLE
feat: migrate from icelake to iceberg-rust

### PR DESCRIPTION
**Pull request by ZENOTME**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://raw.githubusercontent.com/risingwavelabs/risingwave/17af8a747593ebdbfa826691daf75bdab7d14fa0/.github/contributor-license-agreement.txt).

## What's changed and what's your intention?

This PR migrate icelake to iceberg-rust.
After https://github.com/risingwavelabs/iceberg-rust/pull/10, iceberg-rust can support write, so we can migrate all things from icelake to iceberg-rust. 

## Checklist

- [ ] I have written necessary rustdoc comments.
- [ ]  I have added necessary unit tests and integration tests.
- [ ]  I have added test labels as necessary. 
- [ ]  I have added fuzzing tests or opened an issue to track them. 
- [ ]  My PR contains breaking changes. 
- [ ]  My PR changes performance-critical code, so I will run (micro) benchmarks and present the results. 
- [ ]  My PR contains critical fixes that are necessary to be merged into the latest release. 

## Documentation

- [ ]  My PR needs documentation updates. 

<details>
<summary><b>Release note</b></summary>

</details>

### CODE REVIEW

**Comment by ZENOTME** [comment]
We should merge https://github.com/risingwavelabs/risingwave/issues/10 first, and after that, I will rebase this PR and convert it into the normal state.

**Comment by Li0k** [comment]
> We should merge #10 first, and after that, I will rebase this PR and convert it into the normal state.

Is wrong PR link in the description?

**Comment by ZENOTME** [comment]
> > We should merge #10 first, and after that, I will rebase this PR and convert it into the normal state.
> 
> Is wrong PR link in the description?

Thanks! Good catch!🤣

File: src/connector/src/connector_common/iceberg/jni_catalog.rs [hunk 1]
```diff
@@ -88,107 +82,6 @@ pub struct JniCatalog {
...
-                    format!(
-                        "Failed to update iceberg table: {}",
-                        update_table.table_name()
-                    )
-                })?;
-
-            let rust_json_str = jobj_to_str(env, result_json)?;
-
-            let response: CommitTableResponse = serde_json::from_str(&rust_json_str)?;
-
```
Comment (chenzl25) [inline src/connector/src/connector_common/iceberg/jni_catalog.rs:161]: This implementation should be migrated to iceberg-rust JNI as well.

**Review by chenzl25 — APPROVED** [review]
Rest LGTM. Thank you so much for this PR!!!

File: src/connector/src/connector_common/iceberg/jni_catalog.rs [hunk 2]
```diff
@@ -402,8 +312,57 @@ impl CatalogV2 for JniCatalog {
...
+            tracing::info!(
+                "Table metadata location of {} is {}",
+                commit.identifier(),
+                response.metadata_location
+            );
+
+            let table_metadata = response.metadata;
+
+            let file_io = FileIO::from_path(&response.metadata_location)?
+                .with_props(self.config.table_io_configs.iter())
```
Comment (chenzl25) [inline src/connector/src/connector_common/iceberg/jni_catalog.rs:350]: It seems we don't need `self.config` (BaseCatalogConfig is a struct from icelake) anymore, we can simplify it to a map.

**Review by xxchan — APPROVED** [review]
Thanks for the work!!

BTW I think splitting this PR into smaller ones can make it easier to review more carefully:
1. migrate sink to iceberg-rust, without touching legacy code
2. deleting icelake code
3. rename `*_v2` to `*`

But I'm also ok to merge it as long as ci passes since iceberg connector is under heavy development status. 🤪

**Comment by chenzl25** [comment]
Let's move fast and get rid of icelake as soon as possible, because right now we have some known issues can't be resolved in icelake (like statistics). By migrating to iceberg-rust, we can support more complex nested data type for iceberg engine table, google cloud store integration, and so on.

**Comment by ZENOTME** [comment]
Thanks @chenzl25 ! I will send PR to get rid of all icelake later.

**Pull request merged**